### PR TITLE
Update ComboBox-story.js

### PR DESCRIPTION
**Pull request by chrisconnors-ibm**
#### Changelog

**Changed**
 
- removes references to ControlledComboBoxApp & react story item of unsupported, undesired pattern in `packages/react/src/components/ComboBox/ComboBox-story.js`

#### Testing / Reviewing

Inspect [the deploy storybook](https://deploy-preview-6040--carbon-components-react.netlify.app/?path=/story/combobox--default): the ['application-level control for selection' story](https://react.carbondesignsystem.com/?path=/story/combobox--application-level-control-for-selection) should be gone.

### CODE REVIEW

File: packages/react/src/components/ComboBox/ComboBox-story.js [hunk 1]
```diff
@@ -114,13 +81,4 @@ storiesOf('ComboBox', module)
...
-  .add(
-    'application-level control for selection',
-    () => <ControlledComboBoxApp {...props()} />,
-    {
-      info: {
-        text: `Controlled ComboBox example application`,
-      },
-    }
-  );
+  )
```
Comment (emyarod) [inline packages/react/src/components/ComboBox/ComboBox-story.js]:
```suggestion
  );

```

**Review by emyarod — COMMENTED** [review]
it looks like CI is failing because of some formatting issues (running `prettier` should resolve that). not sure if we want to remove the example, but if so, lines 8 and 13 also need to be changed to remove some unused references

**Comment by chrisconnors-ibm** [comment]
>not sure if we want to remove the example, but if so, lines 8 and 13 also need to be changed to remove some unused references

Yeah just trying to remove that one story. caveat: I am not a React Developer.

**Comment by emyarod** [comment]
I'm just unsure if we want to fully remove the story since it demonstrates a technical API. from your perspective which portion of the story is undesired or unsupported?

**Comment by chrisconnors-ibm** [comment]
The pattern of adding a value to the list of suggested values is problematic in a host of ways, and not something we'd want to be seen as approving or encouraging by demonstrating it in common.

The patterns we show here are permission, if not approval to do things. This shouldn't be encouraged at this level.

**Comment by asudoh** [comment]
Thank you for your thoughts @chrisconnors-ibm. I can see a button to add combo box item may not make sense. OTOH there is a general technical requirements in UI components, not necessarily specific to combo box, to cope with a situation where e.g. full combo box data not available at the time of the creation of combo box (i.e. lazy-loaded item data).

@emyarod Here are a couple of ways I can think of wrt how we can move forward on this topic:

1. Making it technical document/unit-test only
2. Add the data via `setTimeout()`

**Comment by chrisconnors-ibm** [comment]
Combo box can't lazy load it's contents, bc you couldn't set the focus to the entry someone types if it's not there. The list in a combo box should never be so long as to require something like that.

You might be thinking of a dropdown with search that constrains the list results but same problem; you can't constrain the results for values that haven't lazy loaded.

In any case that's all well beyond the scope of "what is a combo box". If a react storybook is for tutorials, we should be clear about that, and the tutorials should be both real problems and things we are fully endorsing as patterns.

**Comment by asudoh** [comment]
> Combo box can't lazy load it's contents, bc you couldn't set the focus to the entry someone types if it's not there. The list in a combo box should never be so long as to require something like that.

> You might be thinking of a dropdown with search that constrains the list results but same problem; you can't constrain the results for values that haven't lazy loaded.

You are right - We need to ensure that all combo box items have been there to have them reasonably search-able (without requiring a spinner to be shown), and thus having application lazy-load content as user types a search term (to cope with long list of items) doesn't make sense.

There are some technical use cases where data associated with the dropdown items are not simply available when its UI component is instantiated, due to technical reasons in applications how the code of dropdown and its associated data is loaded and set up. The delay doesn't tend to be long; It's often less than a second, and I don't think the application would make dropdown visible during such "lazy loading" period.

> In any case that's all well beyond the scope of "what is a combo box". If a react storybook is for tutorials, we should be clear about that, and the tutorials should be both real problems and things we are fully endorsing as patterns.

Makes sense, and I think we are on the same page here. I hope my suggestion I put in my earlier comment works for this purpose. If what @emyarod will come up with looks otherwise to you, please don't hesitate to speak up.

**Comment by chrisconnors-ibm** [comment]
Please remove this story from our storybook by approving, or correcting where I've done it wrong and approving, this PR in the meantime.

**Review by emyarod — COMMENTED** [review]
to allow others to resolve the issues in this PR this option in the right sidebar needs to be enabled

image

**Comment by chrisconnors-ibm** [comment]
> to allow others to resolve the issues in this PR this option in the right sidebar needs to be enabled
> 
> image

it is - although maybe you set that?

**Comment by emyarod** [comment]
the checkbox is only visible to the PR author but when I tried pushing to the branch earlier I got a permission denied error

**Review by tw15egan — APPROVED** [review]

**Review by joshblack — APPROVED** [review]

**Pull request merged**